src/Data/Loader/UserLoader.php

```diff
@@ -63,12 +63,18 @@ public function loadKeys( array $keys ) {
 		}
 
 		/**
-		 *
+		 * Populate array with User objects, or null if user does not exist.
 		 */
 		foreach ( $keys as $key ) {
-			$user                   = get_user_by( 'id', $key );
-			$all_users[ $user->ID ] = new User( $user );
+			$user = get_user_by( 'id', $key );
+
+			if ( $user instanceof \WP_User ) {
```

Comment:
Another option here is to do a `try{ } catch{ }` around `User( )` since it throws an exception when you pass it an invalid input. That would look like:
```php
try {
    $all_users[ $user->ID ? $user->ID : $key ] = new User( $user );
catch ( Exception $e ) {
    $all_users[ $key ] = null;
}
```
I think it's important to still return an entry for the `$key` as part of the returned array, since other things downstream will be looking for that, even if it's just a null value.

Reply:
@CodeProKid Thanks for the feedback! Sounds good - I just updated the PR to match your code. So our try/catch will be responsible for catching exceptions for both of these, when `$user` is set to `false`:
- Trying to get property 'ID' of non-object
- `new User()` throwing an exception

Sound good?

Reply:
Yeah, the ternary should deal with the `ID` on `$user` just fine, it's really just for catching the Exception from `User()`.

We **need** the ternary there because the `Trying to get property 'ID' of non-object` error is just that, an error, not an Exception, so the catch wouldn't catch it. 😄

Reply:
So, with the code as above: 

```
$user = get_user_by( 'id', $key );
try {
  $all_users[ $user->ID ? $user->ID : $key ] = new User( $user );
} catch ( \Exception $e ) {
  $all_users[ $key ] = null;
}
```

I get a lot of errors. 

See: 

![Screen Shot 2019-11-08 at 8 51 11 AM](https://user-images.githubusercontent.com/1260765/68489837-1ba34b80-0205-11ea-937f-261791f9a1ce.png)

---- 

But if I use the following code, I do not get said errors:

```
$user = get_user_by( 'id', $key );
if ( $user instanceof \WP_User ) {
  $all_users[ $user->ID ] = new User( $user );
} else if ( ! isset( $all_users[ 0 ] ) ) {
  $all_users[0] = null;
}
```

Reply:
ah, it's because we're using type hinting on the input of that class. That throws an error before it gets to the exception.

We'll have to either use the code @jasonbahl recommended or switch the catch to catch `Throwable`. That would look like:

```php
$user = get_user_by( 'id', $key );
try {
  $all_users[ $user->ID ? $user->ID : $key ] = new User( $user );
} catch ( \Throwable $e ) {
  $all_users[ $key ] = null;
}
```

Reply:
^ This latest from @CodeProKid works for me

Reply:
Actually, now that I'm thinking about it, you can only catch `Throwable`'s in PHP 7+, so we should just go back to the original way.

Reply:
Ah! Good call.